test/recursive_wrapper_move_test.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2017
+// Mikhail Maximov
+//
+// Distributed under the Boost Software License, Version 1.0. (See
+// accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include "boost/config.hpp"
+#include "boost/test/minimal.hpp"
+
+#if !defined(BOOST_NO_CXX14_RETURN_TYPE_DEDUCTION)
```

Comment:
Use `#ifdef __cpp_inheriting_constructors`

Those `__cpp*` macros are now part of the C++ Standard and they are provided by all the compatible compilers https://gcc.gnu.org/projects/cxx-status.html